snapcraft/tests/test_yaml.py

```diff
@@ -399,6 +403,39 @@ def test_config_runtime_environment_ld(self):
                             'Expected LD_LIBRARY_PATH to include "{}"'.format(
                                 expected))
 
+    @unittest.mock.patch('snapcraft.common.get_stagedir')
+    def test_config_stage_environment(self, mock_stagedir):
+        mock_stagedir.return_value = 'foo'
+
+        config = snapcraft.yaml.Config()
+        environment = config.stage_env()
+
+        self.assertTrue('PATH="foo/bin:foo/usr/bin:$PATH"' in environment)
+        self.assertTrue(
+            'LD_LIBRARY_PATH="foo/lib:foo/usr/lib:'
+            'foo/lib/x86_64-linux-gnu:foo/usr/lib/x86_64-linux-gnu:'
+            '$LD_LIBRARY_PATH"' in environment)
+        self.assertTrue(
+            'CFLAGS="-Ifoo/include -Ifoo/usr/include '
+            '-Ifoo/include/x86_64-linux-gnu '
+            '-Ifoo/usr/include/x86_64-linux-gnu $CFLAGS"' in environment)
+        self.assertTrue(
+            'CPPFLAGS="-Ifoo/include -Ifoo/usr/include '
+            '-Ifoo/include/x86_64-linux-gnu '
+            '-Ifoo/usr/include/x86_64-linux-gnu $CPPFLAGS"' in environment)
+        self.assertTrue(
+            'LDFLAGS="-Lfoo/lib -Lfoo/usr/lib -Lfoo/lib/x86_64-linux-gnu '
+            '-Lfoo/usr/lib/x86_64-linux-gnu $LDFLAGS"' in environment)
+        self.assertTrue(
+            'PKG_CONFIG_PATH=foo/lib/pkgconfig:'
+            'foo/lib/x86_64-linux-gnu/pkgconfig:foo/usr/lib/pkgconfig:'
+            'foo/usr/lib/x86_64-linux-gnu/pkgconfig:foo/usr/share/pkgconfig:'
+            'foo/usr/local/lib/pkgconfig:'
+            'foo/usr/local/lib/x86_64-linux-gnu/pkgconfig:'
+            'foo/usr/local/share/pkgconfig:$PKG_CONFIG_PATH' in environment)
```

Comment:
Well I also want to test that the output is correct and no missing colon gets in there. The order is fixed though (and should be as it determines the search order too).